shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/TimeProvider.kt

```diff
@@ -38,15 +42,27 @@ class ServerBasedTimeProvider(private val client: APIClient) : TimeProvider {
 }
 
 class FakeTimeProvider(
-    private val fixedTime: GMTDate = GMTDate(
-        year = 2024, month = Month.MAY, dayOfMonth = 23, hours = 13, minutes = 40, seconds = 0
-    )
+    private val baseTime: GMTDate = GMTDate(
+        year = 2024, month = Month.MAY, dayOfMonth = 23, hours = 12, minutes = 40, seconds = 0
+    ),
+    private val freezeTime: Boolean = false,
 ) : TimeProvider {
-    override val time: StateFlow<GMTDate> = MutableStateFlow(fixedTime)
-    override fun now(): GMTDate = fixedTime
+    private val _time = MutableStateFlow(baseTime)
+    override val time: StateFlow<GMTDate> = _time
+    override fun now(): GMTDate = baseTime
     override suspend fun run(): Nothing {
-        while (true) {
-            delay(Long.MAX_VALUE) // This waits forever on its own
+        if (freezeTime) {
+            awaitCancellation()
+        } else {
+            while (true) {
+                // Progress time at 4x speed for testing
```

Comment:
Please log an issue or at least add `TODO`, so we don't forget to drop it on review

Reply:
This is meant to stay like this, as this `FakeTimeProvider` is unused unless you modify the DI setup in `App.kt`. But if you choose to use it, it'll progress time faster so that it's easier to see how the schedule updates.